### PR TITLE
Answer to a post devas565

### PR DESCRIPTION
**Pull request by paolina92**

### CODE REVIEW

File: assembl/static2/js/app/components/debate/thread/answerForm.jsx [hunk 1]
```diff
@@ -1,36 +1,102 @@
...
+    updateBody(e.target.value);
+  };
+
+  const resetForm = () => {
+    updateAnswerFormStatus(false);
+    updateBodyChars(TEXT_AREA_MAX_LENGTH);
+    updateBody('');
+  };
+
+  const inviteToLogin = () => {
```
Comment (swergas) [inline assembl/static2/js/app/components/debate/thread/answerForm.jsx]: inviteToLogin function is duplicated between answerForm.jsx and topPostForm.jsx. I guess it should be imported from a separate file (and optionnaly call it with configuration parameters).

File: assembl/static2/js/app/reducers/postsReducer.js [hunk 2]
```diff
@@ -29,38 +38,58 @@ export const topPostBody = (state = '', action) => {
...
-  case 'UPDATE_BODY_REMAINING_CHARS':
-    return action.bodyRemainingChars;
+  case 'UPDATE_ANSWER_POST_FORM_STATUS':
+    return action.isAnswerPostFormActive;
   default:
     return state;
   }
 };
 
+export const answerPostBody = (state = '', action) => {
```
Comment (swergas) [inline assembl/static2/js/app/reducers/postsReducer.js]: I would make the same comment as previous PR: https://github.com/assembl/assembl/pull/121#discussion_r126692963 . How about the solution I proposed? (a better name for basicActionFactory could be basicReducerFactory)

File: assembl/static2/js/app/components/debate/thread/answerForm.jsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    updateBody(e.target.value);
+  };
+
+  const resetForm = () => {
+    hideAnswerForm();
+    updateBodyChars(TEXT_AREA_MAX_LENGTH);
+    updateBody('');
+  };
+
+  const inviteToLogin = () => {
```
Comment (paolina92) [inline assembl/static2/js/app/components/debate/thread/answerForm.jsx]: I would like to dispatch an action and store if the user is connected. Then I should move this method in the `main.jsx`. We need to display this modal at several places in the app. What do you think?

File: assembl/static2/js/app/graphql/IdeaWithPosts.graphql [hunk 4]
```diff
@@ -8,7 +8,7 @@ query IdeaWithPosts($lang: String!, $id: ID!) {
       posts(first: 20) {
         edges {
           node {
-            ... on Post {
+            ... on AssemblPost {
```
Comment (swergas) [inline assembl/static2/js/app/graphql/IdeaWithPosts.graphql]: I tried to use an union or interface type instead (defined in graphql/schema.py) but I could not get it to work... Any idea @vincentfretin @maparent @cedricmessiant ?

File: assembl/static2/js/app/reducers/postsReducer.js (hunk 2 above)
Comment (paolina92) [inline assembl/static2/js/app/reducers/postsReducer.js]: I don't know how to write it... No idea. If you have an example...

Comment (swergas) [inline assembl/static2/js/app/reducers/postsReducer.js]: See example I wrote there ;)

File: assembl/static2/js/app/graphql/IdeaWithPosts.graphql (hunk 4 above)
Comment (vincentfretin) [inline assembl/static2/js/app/graphql/IdeaWithPosts.graphql]: Why do want to change that? We don't have PropositionPost on an idea, only posts. Post object type can actually be a sqlalchemy AssemblPost, ImportedPost, Email... Maybe this confused you?

File: assembl/static2/js/app/components/debate/thread/answerForm.jsx (hunk 3 above)
Comment (paolina92) [inline assembl/static2/js/app/components/debate/thread/answerForm.jsx]: Finally, I created a method in a higher level (`utils/utilityManager`) that we can call every where.

File: assembl/static2/js/app/reducers/postsReducer.js (hunk 2 above)
Comment (paolina92) [inline assembl/static2/js/app/reducers/postsReducer.js]: Ok I followed your example

File: assembl/graphql/schema.py [hunk 5]
```diff
@@ -1021,29 +1135,46 @@ def mutate(root, args, context, info):
...
             subject_langstring = langstring_from_input_entries(subject_entries)
             body_langstring = langstring_from_input_entries(body_entries)
-            discussion = models.Discussion.get(discussion_id)
-            new_post = cls(
-                discussion=discussion,
-                subject=subject_langstring,
-                body=body_langstring,
-                creator_id=user_id
-            )
+            if in_reply_to_post: # Is there an unified way to do this?
```
Comment (cedricmessiant) [inline assembl/graphql/schema.py]: @vincentfretin ?

Comment (vincentfretin) [inline assembl/graphql/schema.py]: Yeah, just use a params dict to put parent conditionnaly in it, and create new post with cls(**params)

Comment (vincentfretin) [inline assembl/graphql/schema.py]: Can't you just always use parent=in_reply_to_post here? If in_reply_to_post is None, it should keep the parent field to null.

File: assembl/static2/js/app/graphql/mutations/createPost.graphql [hunk 6]
```diff
@@ -1,15 +1,16 @@
-mutation createPost($ideaId: ID!, $subject: String, $body: String!) {
-  createPost(ideaId:$ideaId, subject: $subject, body: $body) {
+mutation createPost($ideaId: ID!, $subject: String, $body: String!, $parentId: ID) {
+  createPost(ideaId:$ideaId, subject: $subject, body: $body, parentId: $parentId) {
     post {
-      ... on PropositionPost {
+      ... on AssemblPost {
```
Comment (vincentfretin) [inline assembl/static2/js/app/graphql/mutations/createPost.graphql]: If you want to reuse this createPost mutation, you should actually keep the PropositionPost fragment and write the same for Post fragment (you should remove the AssemblPost objecttype in the backend code)

Comment (paolina92) [inline assembl/static2/js/app/graphql/mutations/createPost.graphql]: ok. I'll do

File: assembl/static2/js/app/graphql/IdeaWithPosts.graphql (hunk 4 above)
Comment (vincentfretin) [inline assembl/static2/js/app/graphql/IdeaWithPosts.graphql]: same here, put back Post in the query or you won't get the ImportedPost instances.

**Comment by vincentfretin** [comment]
@swergas @paolina92 I simplified all this in 1c3133388f6db8e5a305562ae4b30f0874aad2e5 and cc816e4e300e948a8d921c0c4e9e698cf51a7d8

**Pull request merged**